rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala

```diff
@@ -28,73 +31,37 @@ import coop.rchain.rspace.spaces.FineGrainedReplayRSpace
 import coop.rchain.shared.StoreType
 import coop.rchain.shared.StoreType._
 import monix.eval.Task
+import monix.execution.Scheduler
 
 import scala.collection.immutable
 
 class Runtime private (
     val reducer: ChargingReducer[Task],
     val replayReducer: ChargingReducer[Task],
-    val space: RhoISpace,
-    val replaySpace: RhoReplayISpace,
+    val space: RhoISpace[Task],
+    val replaySpace: RhoReplayISpace[Task],
     val errorLog: ErrorLog,
     val context: RhoContext,
     val shortLeashParams: Runtime.ShortLeashParams[Task],
     val blockTime: Runtime.BlockTime[Task]
 ) {
   def readAndClearErrorVector(): Vector[Throwable] = errorLog.readAndClearErrorVector()
-  def close(): Unit = {
-    space.close()
-    replaySpace.close()
-    context.close()
-  }
-  def injectEmptyRegistryRoot[F[_]](implicit F: Sync[F]): F[Unit] = {
-    // This random value stays dead in the tuplespace, so we can have some fun.
-    // This is from Jeremy Bentham's "Defence of Usury"
-    val rand = Blake2b512Random(
-      ("there can be no such thing as usury: " +
-        "for what rate of interest is there that can naturally be more proper than another?")
-        .getBytes()
-    )
-    implicit val MATCH_UNLIMITED_PHLOS = matchListPar(Cost(Integer.MAX_VALUE))
-    for {
-      spaceResult <- F.delay(
-                      space.produce(
-                        Registry.registryRoot,
-                        ListParWithRandom(Seq(Registry.emptyMap), rand),
-                        false
-                      )
-                    )
-      replayResult <- F.delay(
-                       replaySpace.produce(
-                         Registry.registryRoot,
-                         ListParWithRandom(Seq(Registry.emptyMap), rand),
-                         false
-                       )
-                     )
-      _ <- spaceResult match {
-            case Right(None) =>
-              replayResult match {
-                case Right(None) => F.unit
-                case Right(Some(_)) =>
-                  F.raiseError(
-                    new SetupError("Registry insertion in replay fired continuation.")
-                  )
-                case Left(err) => F.raiseError(err)
-              }
-            case Right(Some(_)) =>
-              F.raiseError(new SetupError("Registry insertion fired continuation."))
-            case Left(err) => F.raiseError(err)
-          }
-    } yield ()
-  }
+  def close()(
+      implicit scheduler: Scheduler
+  ): Unit =
+    (for {
+      _ <- space.close()
+      _ <- replaySpace.close()
+      _ = context.close()
+    } yield ()).unsafeRunSync
```

Comment:
Is there a particular reason you decided to leave this method as an unpure effect instead of returning a `Task[Unit]`?

Reply:
No reason, my bad